ood-portal-generator/lib/ood_portal_generator/dex.rb

```diff
@@ -158,8 +158,29 @@ def client_name
       @config.fetch(:client_name, "OnDemand")
     end
 
+    def default_secret_path
+      File.join(self.class.config_dir, "ondemand.secret")
+    end
+
+    def generate_secret
+      return default_secret_path if (File.exist?(default_secret_path) && ! File.zero?(default_secret_path))
+
+      secret = SecureRandom.uuid
+      File.open(default_secret_path, "w") { |f| f.write("#{secret}\n") }
```

Comment:
You can do save the line using `FileUtils.chmod` by doing `File.open(default_secret_path, "w", 0600)`

Reply:
This change is made